hs_core/HydroRealtimeSignalProcessor.py

```diff
@@ -0,0 +1,108 @@
+    3. Thus, we want to capture cases in which it is an appropriate instance, and respond. 
+    """
+
+    def hydro_resource_is_present(self, instance): 
+        """ 
+        This returns True if a current resource is now present in Haystack, and False if not. 
+        This -- in turn -- informs us as to whether to update and/or delete the object 
+        """
+        discoverable = SearchQuerySet().filter(id=get_identifier(instance)).count()
+        if discoverable > 0:
+            return True
+        else:
+            return False
+
+    def handle_save(self, sender, instance, **kwargs):
+        """
+        Given an individual model instance, determine which backends the
+        update should be sent to & update the object on those backends.
+        """
+        from hs_core.models import BaseResource
+        from hs_access_control.models import ResourceAccess
+        from hs_labels.models import ResourceLabels
+    
+        if isinstance(instance, BaseResource):
+            if hasattr(instance, 'raccess') and hasattr(instance, 'metadata'): 
+                # work around for failure of super(BaseResource, instance) to work properly.
+                # this always succeeds because this is a post-save object action. 
+                newinstance = BaseResource.objects.get(pk=instance.pk)
+                newsender = BaseResource
+                using_backends = self.connection_router.for_write(instance=newinstance)
+                for using in using_backends:
+                    # if object is public/discoverable or becoming public/discoverable, index it 
+                    if instance.raccess.public or instance.raccess.discoverable: 
+                        try:
+                            index = self.connections[using].get_unified_index().get_index(newsender)
+                            index.update_object(newinstance, using=using)
+                        except NotHandled:
+                            logger.error("Failure: changes to " + str(type(instance)) + " with short_id " + newinstance.short_id+ " not added to SOLR index")
+                    # if object is private or becoming private, delete from index 
+                    elif self.hydro_resource_is_present(newinstance):
+                        try:
+                            index = self.connections[using].get_unified_index().get_index(newsender)
+                            index.remove_object(newinstance, using=using)
+                        except NotHandled:
+                            logger.error("Failure: delete of " + str(type(instance)) + " with short_id " + newinstance.short_id+ " failed.")
+                    else: 
+                        logger.debug("Skipped unnecessary indexing of private " + str(type(instance)) + " with short_id " + newinstance.short_id)
+            else: 
+                logger.debug("Skipped unnecessary indexing of " + str(type(instance)) + " with short_id " + instance.short_id + " (without metadata)")
```

Comment:
@xuezhaokun @alvacouch I am not sure whether debug logging messages on lines 66 and 68 above are useful or not. I feel like we probably do not have to record them and them can probably be removed from production, but I'll defer to you in case these messages provide useful info in the log.